_source/_posts/2024-01-24-spring-boot-authorization.md

```diff
@@ -0,0 +1,569 @@
+package com.example.menu;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.web.SecurityFilterChain;
+
+import static org.springframework.security.config.Customizer.withDefaults;
+
+@Configuration
+public class SecurityConfig {
+
+    @Bean
+    public SecurityFilterChain configure(HttpSecurity http) throws Exception {
+        return http
+                .oauth2ResourceServer(oauth2ResourceServer -> oauth2ResourceServer.jwt(withDefaults()))
+                .build();
+    }
+
+}
+```
+
+Modify `ItemController` by adding the `@CrossOrigin` annotation:
+
+```java
+// src/main/java/com/example/menu/web/ItemController.java
+...
+
+@RestController
+@RequestMapping("/api/menu/items")
+@CrossOrigin(origins = "https://dashboard.whatabyte.app")
+public class ItemController {
+
+...
+```
+
+Restart the server API. Before the sign in, you can create some test users with the Auth0 CLI. This step is optional, you can sign up later from the [Auth0 Universal Login](https://auth0.com/docs/authenticate/login/auth0-universal-login) form, or choose Google social login.
+
+```shell
+auth0 users create
+```
+
+Complete the required information, you will see an output like the lines below:
+
+```text
+Connection Name: Username-Password-Authentication
+ Name: Thomas
+ Email: thomas@email.com
+ Password: ********
+
+=== dev-avup2laz.us.auth0.com user created
+
+  ID          auth0|643ec0e1e671c7c9c5916ed6
+  EMAIL       thomas@email.com
+  CONNECTION  Username-Password-Authentication
+```
+Save the ID for later.
+
+In the dashboard, click the **Sign in** button. The client will redirect you to the [Auth0 Universal Login](https://auth0.com/docs/authenticate/login/auth0-universal-login) page. Sign in with the user you just created. In the left menu, choose **Menu**, and the menu items will display.
+
+{% img blog/spring-boot-authorization/whatabyte-ui.png alt:"WHATABYTE Dashboard" width:"800" %}{: .center-image }
+
+On the top right corner, click **Add Item**, and a pre-populated form will display. Click on **Save** and verify the user request is authorized in the server.
+
+## Role-Based Access Control (RBAC)
+
+If you disable the authentication features in the WHATABYTE client, it will allow you to attempt CRUD operations from the UI, but your API server will reject item modifications as an unauthenticated user.
+
+{% img blog/spring-boot-authorization/whatabyte-unauth.png alt:"WHATABYTE Dashboard Unauthenticated" width:"800" %}{: .center-image }
+
+Besides requiring a user to be authenticated, it is a standard practice to create roles for authorizing write operations.
+
+### Create and assign roles
+
+In the WHATABYTE client settings, re-enable the authentication features, and also enable RBAC. Set `menu-admin` in the _User Role_ text box. Click on **Save**.
```

Comment:
Can we clear this up, what client is this [demo?](https://dashboard.whatabyte.app/home)?